### PR TITLE
feat: add legend

### PR DESCRIPTION
**Pull request by sun0day**
### Description

Several graph UI optimization:

- avoid longpress and drag action trigger `click`
- show the graph node's shape and color meaning
image

### CODE REVIEW

**Comment by antfu** [comment]
I am not sure about `doubleclick`, which might make it harder to discover. I am thinking we could use `pointerdown` and `pointerup`, to calculate the time and moving distance to trigger the event.

**Comment by sun0day** [comment]
Vis wouldn't trigger `pointerdown` and `pointerup`, but I think we could calculate the time between `click` and (`hold` + `dragStart`)

**Comment by antfu** [comment]
To me one of the aspect of good UX is about not guiding "how to do something" but everything work just as expected intuitively. I think dragging and clicking are differentiable actions that making sense. This honestly feels like a bug of `vis` to not providing good enough default. I think we could workaround them for now on our side.

**Comment by sun0day** [comment]
Done

File: src/client/components/Graph.vue [hunk 1]
```diff
@@ -114,15 +107,26 @@ onMounted(() => {
...
     >
-      <span text-sm op50>weight by</span>
-      <RadioGroup
-        v-model="graphWeightMode"
-        name="weight"
-        :options="weightItems"
-      />
+      <div flex="~ gap-2" mb-2>
+        <span text-sm op50>color of</span>
+        <Badge v-for="color of colors" :key="color.type" :class="[`text-[#fff] ${{ vue: 'bg-vue', json: 'bg-json', ts: 'bg-ts', js: 'bg-js', css: 'bg-css', html: 'bg-html', svelte: 'bg-svelte', jsx: 'bg-jsx', tsx: 'bg-tsx' }[color.type]}`]">
```
Comment (sun0day) [inline src/client/components/Graph.vue:114]: BTW, is there a better way to inject a dynamic color class

Comment (antfu) [reply src/client/components/Graph.vue:114]: Just use `:style="{ color }"`, man. Don't go over engineering 😂

Comment (sun0day) [reply src/client/components/Graph.vue]: Sorry bro, I am new to vue&unocss. I need more practice. :(

File: uno.config.ts [hunk 2]
```diff
@@ -17,6 +18,13 @@ export default defineConfig({
       + 'opacity-75 transition duration-200 ease-in-out '
       + 'hover:opacity-100 hover:text-teal-600 '
       + 'text-0.9em h-1.2em',
+    ...colors.reduce((acc, { color, type }) => {
```
Comment (antfu) [inline uno.config.ts:21]: Do we still need this?

File: color.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+export enum COLOR {
```
Comment (antfu) [inline color.ts:1]: I would avoid using enum. Can we just have a plain object?

File: color.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  JS = '#d6cb2d',
+  JSON = '#cf8f30',
+  CSS = '#e6659a',
+  HTML = '#e34c26',
+  SVELTE = '#ff3e00',
+  JSX = '#7d6fe8',
+  TSX = '#7d6fe8',
+}
+
+export const colors = [
```
Comment (antfu) [inline color.ts:13]: This probably can be Object.entries from the color object

File: uno.config.ts (hunk 2 above)
Comment (sun0day) [reply uno.config.ts]: No, I'll remove it

**Pull request merged**